x-pack/plugin/downsample/src/test/java/org/elasticsearch/xpack/downsample/DownsampleActionSingleNodeTests.java

```diff
@@ -825,13 +825,27 @@ public void testDownsampleStats() throws IOException {
             .findFirst()
             .orElseThrow();
 
-        List<Measurement> measurements = plugin.getLongHistogramMeasurement(DownsampleMetrics.LATENCY_SHARD);
-        assertFalse(measurements.isEmpty());
-        for (Measurement measurement : measurements) {
+        final List<Measurement> latencyShardMetrics = plugin.getLongHistogramMeasurement(DownsampleMetrics.LATENCY_SHARD);
+        assertFalse(latencyShardMetrics.isEmpty());
+        for (Measurement measurement : latencyShardMetrics) {
             assertTrue(measurement.value().toString(), measurement.value().longValue() >= 0 && measurement.value().longValue() < 1000_000);
             assertEquals(1, measurement.attributes().size());
             assertThat(measurement.attributes().get("status"), Matchers.in(List.of("success", "failed", "missing_docs")));
         }
+
+        // Total latency gets recorded after reindex and force-merge complete.
+        assertBusy(() -> {
+            final List<Measurement> latencyTotalMetrics = plugin.getLongHistogramMeasurement(DownsampleMetrics.LATENCY_TOTAL);
+            assertFalse(latencyTotalMetrics.isEmpty());
+            for (Measurement measurement : latencyTotalMetrics) {
+                assertTrue(
+                    measurement.value().toString(),
+                    measurement.value().longValue() >= 0 && measurement.value().longValue() < 1000_000
+                );
+                assertEquals(1, measurement.attributes().size());
+                assertThat(measurement.attributes().get("status"), Matchers.in(List.of("success", "failed")));
```

Comment:
Sorry, an afterthought - can't we assert that it is strictly `success` here?

Reply:
This is a very good point, actually. I was expecting more failures, only got `success` here. Added another status for invalid configuration that's also exercised in the test.